inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -468,15 +468,86 @@ struct type_to_type {};
 template <typename typelist>
 struct type_dispatch_impl;
 
+//FIXME: add test for type_dispatch
 template <template<typename ...> class typelist, typename... type>
 struct type_dispatch_impl<typelist<type...>> {
     template <typename result_t, typename default_t, typename type_id_t, typename type_to_id_t, typename type_to_value_t>
     static result_t dispatch(type_id_t type_id, type_to_id_t&& type_to_id, type_to_value_t&& type_to_value, default_t default_value) {
         result_t res = default_value;
 
-        std::initializer_list<int> ({(type_id == type_to_id(type_to_type<type>{}) ? (res = type_to_value(type_to_type<type>{})), 0 : 0)...});
+        bool matched = false;
+        std::initializer_list<int> ({
```

Comment:
i heard that `std::initializer_list` can use dynamic allocation sometimes (depends on implementation) against `std::array`
but i'm not sure 100% - if performance  is critical may be it make sense try `std::array<int, sizeof...(type) >`

Reply:
Ok.I'll try to apply this comment in next PRs-inheritors with refactoring for Resize kernels and will measure the change in performance.